File: 04_Synchronization/15_Berkeley/Berkeley.java
import java.util.*;

public class Berkeley {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n;
        int sum = 0;
        System.out.print("Enter Number of Clocks: ");
        n = sc.nextInt();
        int clock[] = new int[n];
        System.out.println("Enter Clock Times:");
        for(int i = 0; i < n; i++) {
            clock[i] = sc.nextInt();
            sum = sum + clock[i];
        }
        int avg = sum / n;
        System.out.println("\nAverage Clock Time: " + avg);
        System.out.println("\nClock Synchronization:");
        for(int i = 0; i < n; i++) {
            int adjustment = avg - clock[i];
            System.out.println("Clock " + (i + 1) +
                    " Adjustment: " + adjustment);
        }
        System.out.println("\nSynchronized Clock Times:");
        for(int i = 0; i < n; i++) {
            clock[i] = avg;
            System.out.println("Clock " + (i + 1) +
                    " Time: " + clock[i]);
        }
        sc.close();
    }
}

File: 04_Synchronization/15_Berkeley/Berkeley.py
n = int(input("Enter Number of Clocks: "))

clock = []
sum_time = 0

print("Enter Clock Times:")

for i in range(n):
    time = int(input())
    clock.append(time)
    sum_time += time

avg = sum_time // n

print("\nAverage Clock Time:", avg)

print("\nClock Synchronization:")

for i in range(n):
    adjustment = avg - clock[i]
    print("Clock", i + 1, "Adjustment:", adjustment)

print("\nSynchronized Clock Times:")

for i in range(n):
    clock[i] = avg
    print("Clock", i + 1, "Time:", clock[i])